### PR TITLE
remove redundant closures

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

File: examples/app/without_winit.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@ use bevy::winit::WinitPlugin;
 
 fn main() {
     App::new()
-        .add_plugins_with(DefaultPlugins, |group| group.disable::<WinitPlugin>())
+        .add_plugins_with(DefaultPlugins, PluginGroupBuilder::disable)
```
Comment (mockersf) [inline examples/app/without_winit.rs:6]: this one needs to keep the type parameter

**Review by mockersf — APPROVED** [review]

**Comment by cart** [comment]
I'm honestly not sure about most of these. Yes, these closures aren't necessary, but people don't generally think about code like this. The thought process is generally something like this:

1. "This function accepts a closure that accepts an argument, so I'll write that out"
2. "How do I interact with the argument to produce the desired output. What function should I call?"

(2) often involves using autocomplete on the argument to discover the proper method.

The style here isn't a natural part of that process. It feels like adding a "step 3" to the process of "how do i compress this". I personally don't think thats worth it. This type of compression actually makes these things harder to understand imo.

**Comment by danieleades** [comment]
> I'm honestly not sure about most of these. Yes, these closures aren't necessary, but people don't generally think about code like this. The thought process is generally something like this:
> 
> 1. "This function accepts a closure that accepts an argument, so I'll write that out"
> 2. "How do I interact with the argument to produce the desired output. What function should I call?"
> 
> (2) often involves using autocomplete on the argument to discover the proper method.
> 
> The style here isn't a natural part of that process. It feels like adding a "step 3" to the process of "how do i compress this". I personally don't think thats worth it. This type of compression actually makes these things harder to understand imo.  

I guess its a matter of taste. Coming from more imperative languages originally I had the same thinking. The more time I spend working with functional approaches the more I think 'mapping' through a named function is a better fit for the mental model.

I don't think this change is totally clear cut, which is why I split this PR out from #2934

**Comment by cart** [comment]
Yeah definitely a matter of opinion. I prefer to keep these as they are for now, so I'm closing this out. But feel free to start a github discussion if you feel strongly about it. If theres relative consensus, we can reconsider.

**Pull request closed without merging**